x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/use_experimental_feature_fields_transform.ts

```diff
@@ -7,39 +7,16 @@
 
 import { useCallback } from 'react';
 import type { DefineStepRule } from '../../../../detections/pages/detection_engine/rules/types';
-import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
-import { isThreatMatchRule } from '../../../../../common/detection_engine/utils';
 
 /**
  * transforms  DefineStepRule fields according to experimental feature flags
  */
 export const useExperimentalFeatureFieldsTransform = <T extends Partial<DefineStepRule>>(): ((
   fields: T
 ) => T) => {
-  const isAlertSuppressionForIndicatorMatchRuleEnabled = useIsExperimentalFeatureEnabled(
-    'alertSuppressionForIndicatorMatchRuleEnabled'
-  );
-
-  const transformer = useCallback(
-    (fields: T) => {
-      const isIndicatorMatchSuppressionDisabled = isThreatMatchRule(fields.ruleType)
-        ? !isAlertSuppressionForIndicatorMatchRuleEnabled
-        : false;
-      // reset any alert suppression values hidden behind feature flag
-      if (isIndicatorMatchSuppressionDisabled) {
-        return {
-          ...fields,
-          groupByFields: [],
-          groupByRadioSelection: undefined,
-          groupByDuration: undefined,
-          suppressionMissingFields: undefined,
-        };
-      }
-
-      return fields;
-    },
-    [isAlertSuppressionForIndicatorMatchRuleEnabled]
-  );
+  const transformer = useCallback((fields: T) => {
+    return fields;
+  }, []);
 
   return transformer;
```

Comment:
yes, it just returns `fields` on 18th line without any transformations.

As there are no feature flags, so it just return passed fields property. It can be removed altogether, but I just removed threat match feature flag.
It will be used for other rule types FF, as EQL